File: 283-MoveZeroes/283-MoveZeroes.py
# Last updated: 28/05/2025, 14:52:41
class Solution:
    def calcEquation(self, equations: List[List[str]], values: List[float], queries: List[List[str]]) -> List[float]:
        graph= defaultdict(dict)

        for(u,v),val in zip(equations,values):
            graph[u][u]= graph[v][v]=1
            graph[u][v]= val
            graph[v][u]= 1/val

        for k in graph:
            for i in graph[k]:
                for j in graph[k]:
                    graph[i][j]= graph[i][k]*graph[k][j] if i!=j else 1
        ans=[]

        for u,v in queries:
            ans.append(graph[u].get(v,-1))
        return ans


        

File: 283-MoveZeroes/283-MoveZeroes.cpp
// Last updated: 28/05/2025, 09:59:01
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        k=k%nums.size();
        reverse(nums.begin(),nums.begin()+ (nums.size()-k));
        reverse(nums.begin()+ (nums.size()-k), nums.end());
        reverse(nums.begin(),nums.end());

        
    }
};